studio/hooks/misc/useTableRowsPrefetch.ts

```diff
@@ -0,0 +1,37 @@
+import { SupaTable } from 'components/grid'
+import { formatFilterURLParams, formatSortURLParams } from 'components/grid/SupabaseGrid.utils'
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { useTableRowsPrefetch as _useTableRowsPrefetch } from 'data/table-rows/table-rows-query'
+import { useUrlState } from 'hooks'
+import { useCallback } from 'react'
+
+/**
+ * useTableRowsPrefetch is a wrapper around the base useTableRowsPrefetch that prefills the sort and filter params
+ */
+function useTableRowsPrefetch() {
```

Comment:
quick question - we can't merge this with what's in `table-rows/table-rows-query`?

Reply:
I'm trying to keep everything in the `data` folder only related to data fetching. No dependencies on any other parts of the app (like the `useProjectContext()` for example). So maybe the issue here is file naming/file location? Can you think of a better place for this?

Reply:
think this is fine for now really - the other possible alternatives are
- shift the sort, filter, project context stuff into TableEditorLayout, then we can just use the RQ hook directly
  - i think this will be consistent with how we're using RQ around the app too since a couple of the RQ hooks need the project context
- Shift this into the TableEditorLayout folder under xxx.utils.ts, since this is very specific to the TableEditor
- renaming it to useTableRowsPrefetchWrapper

what do you think?

Reply:
I think yes – I've gone or the second two suggestions.